Counterexamples/HomogeneousPrimeNotPrime.lean

```diff
@@ -0,0 +1,218 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Eric Wieser, Jujian Zhang
+
+! This file was ported from Lean 3 source module homogeneous_prime_not_prime
+! leanprover-community/mathlib commit 328375597f2c0dd00522d9c2e5a33b6a6128feeb
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.RingTheory.GradedAlgebra.HomogeneousIdeal
+import Mathlib.Data.ZMod.Basic
+import Mathlib.Tactic.DeriveFintype
+
+/-!
+# A homogeneous prime that is homogeneously prime but not prime
+
+In `src/ring_theory/graded_algebra/radical.lean`,  we assumed that the underline grading is indexed
+by a `linear_ordered_cancel_add_comm_monoid` to prove that a homogeneous ideal is prime if and only
+if it is homogeneously prime. This file is aimed to show that even if this assumption isn't strictly
+necessary, the assumption of "being cancellative" is. We construct a counterexample where the
+underlying indexing set is a `linear_ordered_add_comm_monoid` but is not cancellative and the
```

Comment:
Did you run `fix-comments` on this file?

Reply:
Oops, now I have